[Flaky Test] //pkg/kubelet/cm/devicemanager/go_default_test:run_2_of_2

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

**Which jobs are flaking**:
bazel-test-master

**Which test(s) are flaking**:
//pkg/kubelet/cm/devicemanager/go_default_test:run_2_of_2

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-blocking#bazel-test-master

**Reason for failure**:
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //pkg/kubelet/cm/devicemanager:go_default_test
-----------------------------------------------------------------------------
I1127 05:25:45.047731  215379 device_plugin_stub.go:142] Starting to serve on /tmp/mock1606454745037022410.sock
I1127 05:25:45.057478  215379 device_plugin_stub.go:142] Starting to serve on /tmp/mock1606454745054717131.sock
I1127 05:25:45.067977  215379 device_plugin_stub.go:238] ListAndWatch
E1127 05:25:45.070649  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin mock with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.079328  215379 device_plugin_stub.go:142] Starting to serve on /tmp/mock1606454745070838053.sock
I1127 05:25:45.087164  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.088887  215379 device_plugin_stub.go:272] Allocate, &AllocateRequest{ContainerRequests:[]*ContainerAllocateRequest{&ContainerAllocateRequest{DevicesIDs:[ADeviceId],},},}
E1127 05:25:45.092947  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin mock with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.096696  215379 device_plugin_stub.go:142] Starting to serve on /tmp/mock1606454745093025638.sock
I1127 05:25:45.130907  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.132196  215379 device_plugin_stub.go:259] GetPreferredAllocation, &PreferredAllocationRequest{ContainerRequests:[]*ContainerPreferredAllocationRequest{&ContainerPreferredAllocationRequest{AvailableDeviceIDs:[],MustIncludeDeviceIDs:[],AllocationSize:-1,},},}
E1127 05:25:45.135911  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin mock with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.136291  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
I1127 05:25:45.136808  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
W1127 05:25:45.137220  215379 manager.go:594] Failed to retrieve checkpoint for "kubelet_internal_checkpoint": checkpoint is not found
I1127 05:25:45.140778  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin095277315/device-plugin.sock
I1127 05:25:45.143199  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
W1127 05:25:45.143567  215379 manager.go:594] Failed to retrieve checkpoint for "kubelet_internal_checkpoint": checkpoint is not found
I1127 05:25:45.144854  215379 plugin_manager.go:114] Starting Kubelet Plugin Manager
I1127 05:25:45.150082  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin819249798/device-plugin.sock
I1127 05:25:45.151700  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
W1127 05:25:45.151921  215379 manager.go:594] Failed to retrieve checkpoint for "kubelet_internal_checkpoint": checkpoint is not found
I1127 05:25:45.156950  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock
I1127 05:25:45.157052  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.164156  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.169437  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.186903  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.new
I1127 05:25:45.186998  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.189419  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.196192  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.207455  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.third
I1127 05:25:45.207551  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.216424  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.220033  215379 device_plugin_stub.go:238] ListAndWatch
E1127 05:25:45.228630  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.228776  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.228864  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
E1127 05:25:45.229176  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.234940  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock
I1127 05:25:45.235032  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.237502  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.243370  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.252818  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.new
I1127 05:25:45.252915  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.256615  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.265191  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.273465  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.third
I1127 05:25:45.273555  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.276665  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.281857  215379 device_plugin_stub.go:238] ListAndWatch
E1127 05:25:45.287747  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.287867  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.289423  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.290027  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
I1127 05:25:45.293746  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock
I1127 05:25:45.293846  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.297271  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.304811  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.321498  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.new
I1127 05:25:45.321610  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.326368  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.334582  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.345882  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.third
I1127 05:25:45.345977  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.351284  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.358952  215379 device_plugin_stub.go:238] ListAndWatch
E1127 05:25:45.368886  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.369295  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.370467  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
E1127 05:25:45.370766  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.375061  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock
I1127 05:25:45.375146  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.379000  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.383428  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.391178  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.new
I1127 05:25:45.391281  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.394611  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.399425  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.407200  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin077505581/device-plugin.sock.third
I1127 05:25:45.407298  215379 device_plugin_stub.go:191] Deprecation file not found. Invoke registration
I1127 05:25:45.414612  215379 manager.go:408] Got registration request from device plugin with resource name "fake-domain/resource"
I1127 05:25:45.418845  215379 device_plugin_stub.go:238] ListAndWatch
E1127 05:25:45.424384  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.424528  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
E1127 05:25:45.425051  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Unavailable desc = transport is closing
I1127 05:25:45.427390  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
W1127 05:25:45.427660  215379 manager.go:594] Failed to retrieve checkpoint for "kubelet_internal_checkpoint": checkpoint is not found
I1127 05:25:45.430337  215379 plugin_manager.go:114] Starting Kubelet Plugin Manager
I1127 05:25:45.432482  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin151853188/device-plugin.sock
I1127 05:25:45.436246  215379 device_plugin_stub.go:164] GetInfo
E1127 05:25:45.439735  215379 goroutinemap.go:150] Operation for "/tmp/device_plugin151853188/server.sock" failed. No retries permitted until 2020-11-27 05:25:45.937941568 +0000 UTC m=+1.094479632 (durationBeforeRetry 500ms). Error: "RegisterPlugin error -- failed to get plugin info using RPC GetInfo at socket /tmp/device_plugin151853188/server.sock, err: rpc error: code = Unimplemented desc = unknown service pluginregistration.Registration"
I1127 05:25:45.442921  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:45.452971  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin151853188/device-plugin.sock.new
E1127 05:25:46.433946  215379 endpoint.go:107] listAndWatch ended unexpectedly for device plugin fake-domain/resource with error rpc error: code = Canceled desc = grpc: the client connection is closing
I1127 05:25:46.446896  215379 device_plugin_stub.go:164] GetInfo
E1127 05:25:46.447263  215379 goroutinemap.go:150] Operation for "/tmp/device_plugin151853188/server.sock" failed. No retries permitted until 2020-11-27 05:25:47.447080785 +0000 UTC m=+2.603618893 (durationBeforeRetry 1s). Error: "RegisterPlugin error -- failed to get plugin info using RPC GetInfo at socket /tmp/device_plugin151853188/server.sock, err: rpc error: code = Unimplemented desc = unknown service pluginregistration.Registration"
I1127 05:25:46.451383  215379 device_plugin_stub.go:164] GetInfo
I1127 05:25:46.455485  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:46.461114  215379 device_plugin_stub.go:238] ListAndWatch
I1127 05:25:46.468597  215379 device_plugin_stub.go:142] Starting to serve on /tmp/device_plugin151853188/device-plugin.sock.third
--- FAIL: TestDevicePluginReRegistrationProbeMode (1.04s)
    manager_test.go:223: 
        	Error Trace:	manager_test.go:223
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestDevicePluginReRegistrationProbeMode
        	Messages:   	Devices of previous registered should be removed
I1127 05:25:46.474428  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
E1127 05:25:46.505334  215379 manager.go:544] unexpected: unhealthyDevices and endpoints are out of sync
I1127 05:25:46.525706  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
I1127 05:25:46.526291  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 5532d3cf-972c-4b23-acb7-aedbd99eac0d container name:  
I1127 05:25:46.526526  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 5532d3cf-972c-4b23-acb7-aedbd99eac0d container name:  
I1127 05:25:46.534916  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: e06661bc-2e81-4ae1-b671-e9f97e69bbea container name:  
I1127 05:25:46.541748  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
I1127 05:25:46.542045  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 9b379718-f72b-4c05-9bff-d54b4483f613 container name:  f78dc3c6-9cc4-456e-8cd8-88e90b836ffa
I1127 05:25:46.547467  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 9b379718-f72b-4c05-9bff-d54b4483f613 container name:  57793332-2b3e-4cbf-b01f-ee8e3dd305bf
I1127 05:25:46.553218  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 9b379718-f72b-4c05-9bff-d54b4483f613 container name:  80783646-5c50-4bbb-b64f-14f29d1351ce
I1127 05:25:46.559343  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 9b379718-f72b-4c05-9bff-d54b4483f613 container name:  7e774d21-0355-43f9-908f-aef222512c9b
I1127 05:25:46.567402  215379 fake_topology_manager.go:30] [fake topologymanager] NewFakeManager
I1127 05:25:46.567627  215379 fake_topology_manager.go:35] [fake topologymanager] GetAffinity pod: 1dc10f8e-b8e6-4afb-9aae-9056682fbb9c container name:  
I1127 05:25:46.580479  215379 topology_hints.go:46] [devicemanager] Resource 'testdevice' does not have a topology preference
E1127 05:25:46.581214  215379 topology_hints.go:70] [devicemanager] Unable to generate topology hints: requested number of devices unavailable for 'testdevice': requested: 6, available: 4
I1127 05:25:46.581370  215379 topology_hints.go:61] [devicemanager] Regenerating TopologyHints for resource 'testdevice' already allocated to (pod _(fakePod), container fakeContainer)
E1127 05:25:46.581525  215379 topology_hints.go:57] [devicemanager] Resource 'testdevice' already allocated to (pod _(fakePod), container fakeContainer) with different number than request: requested: 4, allocated: 2
E1127 05:25:46.581748  215379 topology_hints.go:57] [devicemanager] Resource 'testdevice' already allocated to (pod _(fakePod), container fakeContainer) with different number than request: requested: 2, allocated: 4
I1127 05:25:46.596442  215379 topology_hints.go:96] [devicemanager] Resource 'testdevice' does not have a topology preference
I1127 05:25:46.596945  215379 topology_hints.go:111] [devicemanager] Regenerating TopologyHints for resource 'testdevice' already allocated to (pod _(fakePod))
E1127 05:25:46.597171  215379 topology_hints.go:119] [devicemanager] Unable to generate topology hints: requested number of devices unavailable for 'testdevice': requested: 6, available: 4
I1127 05:25:46.597302  215379 topology_hints.go:111] [devicemanager] Regenerating TopologyHints for resource 'testdevice' already allocated to (pod _(fakePod))
E1127 05:25:46.597438  215379 topology_hints.go:107] [devicemanager] Resource 'testdevice' already allocated to (pod _(fakePod)) with different number than request: requested: 4, allocated: 2
E1127 05:25:46.597550  215379 topology_hints.go:107] [devicemanager] Resource 'testdevice' already allocated to (pod _(fakePod)) with different number than request: requested: 2, allocated: 4
FAIL
```

**Anything else we need to know**:
- links to go.k8s.io/triage appreciated
Triage shows nothings (seems to not have updated data), but there are multiple failures in the job bazel-test-master.

- links to specific failures in spyglass appreciated

  * https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1332184419200405504
  * https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1330993756861632512

<!-- Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance! -->

@kubernetes/ci-signal

Digging further as the step  
//staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/go_default_test:run_2_of_2 
has another recent test fail ... 

So on the two runs reported by @ScrapCodes 
prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1332184419200405504
prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1330993756861632512

[1332184419200405504](prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1332184419200405504) has 2 test failures
[FAIL: TestAdmit/match_dry_run_side_effects_None (0.02s)](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1332184419200405504#1:build-log.txt%3A307)
[FAIL: TestDevicePluginReRegistrationProbeMode (1.04s)](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1332184419200405504#1:build-log.txt%3A180)


[1330993756861632512](prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1330993756861632512) has 1 test failure
[FAIL: TestDevicePluginReRegistrationProbeMode (2.03s)](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1330993756861632512#1:build-log.txt%3A265)

And the most recent failure in this area is a TestAdmit/match test failure but on a different test.
[--- FAIL: TestAdmit/match_&_allow_(url) (0.01s)](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-kubernetes-bazel-test-master/1329133117591523328#1:build-log.txt%3A182)

--- a/pkg/kubelet/cm/devicemanager/manager_test.go
+++ b/pkg/kubelet/cm/devicemanager/manager_test.go
@@ -282,8 +282,8 @@ func setup(t *testing.T, devs []*pluginapi.Device, callback monitorCallback, soc
 
 func setupInProbeMode(t *testing.T, devs []*pluginapi.Device, callback monitorCallback, socketName string, pluginSocketName string) (Manager, <-chan interface{}, *Stub, pluginmanager.PluginManager) {
 	m, updateChan := setupDeviceManager(t, devs, callback, socketName)
-	pm := setupPluginManager(t, pluginSocketName, m)
 	p := setupDevicePlugin(t, devs, pluginSocketName)
+	pm := setupPluginManager(t, pluginSocketName, m)
 	return m, updateChan, p, pm
 }
 

--- a/pkg/kubelet/pluginmanager/reconciler/reconciler.go
+++ b/pkg/kubelet/pluginmanager/reconciler/reconciler.go
@@ -100,7 +100,11 @@ func (rc *reconciler) getHandlers() map[string]cache.PluginHandler {
 	rc.RLock()
 	defer rc.RUnlock()
 
-	return rc.handlers
+	var copyHandlers = make(map[string]cache.PluginHandler)
+	for pluginType, handler := range rc.handlers {
+		copyHandlers[pluginType] = handler
+	}
+	return copyHandlers
 }
 
 func (rc *reconciler) reconcile() {

